app/services/mentor_at_school_periods/finish.rb

```diff
@@ -12,6 +13,14 @@ def finish_existing_at_school_periods!
         period.update!(finished_on:)
         period.mentorship_periods.ongoing_on(finished_on).update!(finished_on:)
         period.training_periods.ongoing_on(finished_on).update!(finished_on:)
```

Comment:
These can now use `MentorshipPeriods::Finish` and `TrainingPeriods::Finish` instead, which will write events for them too.